analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnFloatingPoint.cs

```diff
@@ -83,6 +85,11 @@ private static void CheckEquality(SonarSyntaxNodeReportingContext context)
         }
     }
 
+    // Only checks for type and member accessed. No need to check that proposedMethod exists in type because:
+    // - both special members and corresponding `IsXXX()` methods exist in double and float since .NET 1.1
+    // - both special members and corresponding `IsXXX()` methods exist in Half since its introduction, in .NET 5
+    // - while NFloat has been introduced in .NET 6, special members and corresponding IsXXX() methods have been introduced at the same time, in .NET 7
+    // - no other floating point typeswith prevision issues exist before .NET 7, and any new type after has to conform to `IFloatingPointIeee754`
```

Comment:
We don't need this in the codebase. It would be suitable to put it in the PR description and it's good you did the research. 
```suggestion
```